uplink/storage/streams/store.go

```diff
@@ -437,19 +437,24 @@ func (s *streamStore) Delete(ctx context.Context, path Path, pathCipher storj.Ci
 		return err
 	}
 
+	var errlist errs.Group
 	for i := 0; i < int(numberOfSegments(&stream, &streamMeta)-1); i++ {
 		currentPath, err := createSegmentPath(ctx, int64(i), path.Bucket(), encPath)
 		if err != nil {
-			return err
+			errlist.Add(err)
+			continue
 		}
 
 		err = s.segments.Delete(ctx, currentPath)
 		if err != nil {
-			return err
+			errlist.Add(err)
+			continue
 		}
 	}
 
-	return s.segments.Delete(ctx, lastSegmentPath)
+	errlist.Add(s.segments.Delete(ctx, lastSegmentPath))
+
+	return errlist.Err()
```

Comment:
We'll return errors to the client, however, the segments may be deleted despite the errors.
I'm wondering how we can inform of that to the callers.

Reply:
yeah, this is problematic. maybe we can make different classes for errors and distinguish metainfo errors from SN errors but as I mentioned earlier we should improve the whole process in general.